js/web/lib/wasm/jsep/webgpu/ops/gather-nd.ts

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { DataType } from '../../../wasm-common';
+import { TensorView } from '../../tensor-view';
+import { ShapeUtil } from '../../util';
+import { AttributeWithCacheKey } from '../attribute-with-cache-key';
+import { ComputeContext, ProgramUniform } from '../types';
+
+import { createTensorShapeVariables, inputVariable, outputVariable, ShaderHelper, UniformsArrayType } from './common';
+
+export interface GatherNDAttributes extends AttributeWithCacheKey {
+  readonly batchDims: number;
+}
+
+const computeSliceOffsets = (
+  context: ComputeContext,
+  indicesData: TensorView,
+  sizesFromSliceDimsData: number[],
+  batchDims: number,
+  inputDims: readonly number[],
+  numSlices: number,
+  numSlicesPerBatch: number,
+  inputBatchStride: number,
+  numSliceDims: number,
+) => {
+  const programUniforms: ProgramUniform[] = [
+    { type: DataType.uint32, data: numSlices },
+    { type: DataType.uint32, data: batchDims },
+    { type: DataType.uint32, data: inputDims },
+    { type: DataType.uint32, data: sizesFromSliceDimsData },
+    { type: DataType.uint32, data: numSlicesPerBatch },
+    { type: DataType.uint32, data: inputBatchStride },
+    { type: DataType.uint32, data: numSliceDims },
+  ];
+
+  const outputShape = [numSlices];
+  programUniforms.push(...createTensorShapeVariables(indicesData.dims, outputShape));
+
+  const getShaderSource = (shaderHelper: ShaderHelper) => {
+    const indices = inputVariable('indices_data', indicesData.dataType, indicesData.dims.length);
+    const output = outputVariable('input_slice_offsets_data', DataType.int64, 1, 1);
```

Comment:
My previous comments may have confused you. I was trying to say that using type u32 instead of any 64bit integer type (int64/uint64) is better for the output.